docs/source/conf.py

```diff
@@ -44,8 +45,14 @@
 }
 autoclass_content = "both"
 
-nbsphinx_allow_errors = True
-nbsphinx_execute = "never"
+nb_execution_mode = "auto"
+nbsphinx_allow_errors = False
```

Comment:
We might get blocked by some annoying sphinx issues if we set this to False.

Reply:
Potentially. However, it'll also allow us to identify issues within notebooks quickly. We should decide whether this tradeoff is worth it. In my opinion, occasional Sphinx issues are OK if it means we never have notebook issues. Keen to hear your thoughts.

Reply:
Yeah ok, let's try this! We can easily explore this option and see and if setting False become a significant problem or not.

Reply:
Great! I'll address these comments (unresolved them so I don't forget) and finish this PR next week.